src/state.rs

```diff
@@ -1,17 +1,34 @@
-use cosmwasm_std::Response;
+use cosmwasm_std::{Response, Addr, Timestamp};
 use cw_storage_plus::Item;
-
-use crate::{data::DealMetadata, error::ContractError, execute::Context, msg::InstantiateMsg};
-
+use crate::{data::{DealMetadata, Deal}, error::ContractError, execute::Context, msg::InstantiateMsg};
 use super::data::Config;
 
 pub const CONFIG: Item<Config> = Item::new("config");
 pub const DEAL_METADATA: Item<DealMetadata> = Item::new("deal_metadata");
+pub const DEAL: Item<Deal> = Item::new("deal");
 
 /// Top-level initialization of contract state
 pub fn init_contract(
-    _ctx: Context,
-    _msg: InstantiateMsg,
+    ctx: Context,
+    msg: InstantiateMsg,
 ) -> Result<Response, ContractError> {
+    let metadata = DealMetadata {
+        name: msg.name,
+        description: msg.description,
+        image_url: None, // Placeholder
+    };
+    
+    let deal = Deal {
+        id: "initial_deal".to_string(), // Placeholder
+        created_at: Timestamp::from_seconds(ctx.env.block.time.seconds()),
```

Comment:
The `Timestamp::from_seconds` isn't necessary here. You can store the `Timestamp` directly, so `created_at: env.block.time` would be fine. This preserves the nanosecond precision. However, see my comment below. This comment here is in regards to saving timestamps in general